fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -943,6 +951,67 @@ default RecordCursor<FDBIndexedRecord<M>> scanIndexRecords(@Nonnull final String
                 scanProperties.getExecuteProperties().getState());
     }
 
+    /**
+     * Scan the records pointed to by an index. This version of the method can choose between various scan options as
+     * determined by the {@link RecordQueryPlannerConfiguration.IndexFetchMethod} parameter:
+     * <UL>
+     *     <LI>SCAN_AND_FETCH: Scan the index and fetch each record individually</LI>
+     *     <LI>USE_REMOTE_FETCH: Use FDB's remote fetch feature to scan the index and fetch the records in one call</LI>
+     *     <LI>USE_REMOTE_FETCH_WITH_FALLBACK: Use FDB's remote fetch feature but fall back to SCAN_AND_FETCH in case of failure</LI>
+     * </UL>
+     * This method can only be used with {@link IndexScanRange} scan bounds
+     *
+     * @param indexName the name of the index
+     * @param fetchMethod teh fetch method to use when getting records from the index
+     * @param scanBounds the range of the index to scan
+     * @param commonPrimaryKey the common primary key for the records that would be returned, only required for USE_REMOTE_FETCH and USE_REMOTE_FETCH_WITH_FALLBACK
+     * @param continuation any continuation from a previous scan
+     * @param orphanBehavior how the iteration process should respond in the face of entries in the index for which there is no associated record
+     * @param scanProperties skip, limit and other scan properties
+     * @return a cursor that return records pointed to by the index
+     */
+    @API(API.Status.EXPERIMENTAL)
+    @Nonnull
+    default RecordCursor<FDBIndexedRecord<M>> scanIndexRecords(@Nonnull final String indexName,
+                                                               @Nonnull final RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod,
```

Comment:
Would it make sense to move this `enum` into this class? It seems kind of abstraction-stretching to depend on something supposedly belonging to the query planner.

Reply:
Sure - I can make the enum standalone, so the same type can be used from both query and scan.

Reply:
(Unless you find that inappropriate)

Reply:
Yes, that was the idea. It seems reasonable for the planner config to have store's scan setting, just not the other way around.

Reply:
Done.